src/main/kotlin/com/tealcube/minecraft/bukkit/mythicdrops/identification/IdentificationInventoryDragListener.kt

```diff
@@ -92,6 +93,21 @@ internal class IdentificationInventoryDragListener(
             return
         }
 
+        var preIdentificationEvent = PreIdentificationEvent(targetItem, tier, player)
+        Bukkit.getPluginManager().callEvent(preIdentificationEvent)
+
+        if (preIdentificationEvent.isCancelled) {
+            Log.debug("identificationEvent.isCancelled")
+            player.sendMythicMessage(
+                    settingsManager.languageSettings.identification.failure,
+                    "%reason%" to "identification event is cancelled"
+            )
+            return
+        }
+        else if (preIdentificationEvent.isModified) {
+            tier = preIdentificationEvent.tier
```

Comment:
Instead of changing `tier` here, can you pass `preIdentificationEvent.tier` into the drop builder?